compiler/rustc_mir_build/src/thir/pattern/deconstruct_pat.rs

```diff
@@ -630,9 +640,9 @@ pub(super) enum Constructor<'tcx> {
     /// Ranges of integer literal values (`2`, `2..=5` or `2..5`).
     IntRange(IntRange),
     /// Ranges of floating-point literal values (`2.0..=5.2`).
-    FloatRange(ty::Const<'tcx>, ty::Const<'tcx>, RangeEnd),
+    FloatRange(mir::ConstantKind<'tcx>, mir::ConstantKind<'tcx>, RangeEnd),
```

Comment:
:thinking: i guess using mir constant here is fine :thinking: float ranges are weird anyways